certora/specs/AllocationMorphoVaultV1Adapter.spec

```diff
@@ -25,17 +25,19 @@ methods {
     function Math.mulDiv(uint256 x, uint256 y, uint256 denominator) internal returns (uint256) => mulDivSummary(x, y, denominator);
     function _.supplyShares(address, MorphoHarness.Id id, address user) internal => summarySupplyShares(id, user) expect uint256;
 
+    // Avoids having to prove the invariant: vault v1 always calls Morpho with markets where the loan token is the same as the vault asset.
+    // This invariant is a corollary of MarketInteractions and ConsistentState.enabledHasConsistentAsset.
```

Comment:
> After that (and https://github.com/morpho-org/vault-v2/pull/857), DISPATCHER(true) is only used for tokens (which is sound because we dispatch to tokens that cover "all the supported behaviors")

It's a sane assumption to make, but it's still an assumption that can hide bugs (so I wouldn't say that it's sound). This is because we sometimes rely on the code to call the appropriate address for the tokens. The best example is the comment above: the dispatcher assumption could make us miss that we didn't do the appropriate checks in vault v1